lib/cErrors.ml

```diff
@@ -126,7 +133,7 @@ let print_gen ~anomaly (e, info) =
 
 (** The standard exception printer *)
 let iprint (e, info) =
-  print_gen ~anomaly:true (e,info) ++ print_backtrace info
+  print_gen ~anomaly:true (e,info) ++ print_quickfix info ++ print_backtrace info
```

Comment:
should use register_additional_error_info instead of additional adhoc handling

Reply:
Thanks for the hint, I did not know that API existed

Reply:
also I don't know if using exninfo is the right way to do it, maybe a table of `exn -> quickfix option` functions would be better
then in cwarnings we would have `type w = W : 'a tag * quickfix option * 'a -> w` instead of the current `type w = W : 'a tag * 'a -> w`, or maybe delay the `'a -> quickfix` computation as is done for printing

Reply:
Why not a `('a, 'qc) With_quickfix_info.t = { 'a; qc_info : unit -> 'qc }` wrapper, that would allow the upper layers to apply the quickfixes later? I dunno, there are several designs here.

Reply:
> also I don't know if using exninfo is the right way to do it

Why not? It is a general mechanism to enrich an exception and is implemented using a table. What are the advantages of a quickfix-specific table?

Reply:
Hum, I see what you mean by fragile, but I'm not so sure a map `exn -> whatever` can do much better. After all if one catches an exception he may as well throw another one (with or without the same info attached). I'd rather avoid 2 capture/reraise APIs.

Also note that Loc.t is the info as well, and although not critical it is quite relevant to the user.

Reply:
>I'd rather avoid 2 capture/reraise APIs.

I am not proposing a capture API, I propose that quickfix is like printing

Reply:
Ok, so you propose that each error with a quickfix should have a dedicated exception (not a UserError) and that exception should carry the quickfix. For each of such errors one registers a function to Quickfix.t, that is used by the toplevel in order to get the quickfix data out of the exception.

I think I get it now.

Reply:
Yes.
And then the function can be trivial (`MyExn (_, qf) -> qf`) or less trivial if we want to delay computation

Reply:
done